concrete/src/Entity/User/LoginAttempt.php

```diff
@@ -0,0 +1,94 @@
+<?php
+
+namespace Concrete\Core\Entity\User;
+use Doctrine\ORM\Mapping as ORM;
+
+/**
+ * @ORM\Entity(repositoryClass="Concrete\Core\Entity\User\LoginAttemptRepository")
+ * @ORM\Table(
+ *     name="UserLoginAttempts"
+ * )
+ */
+class LoginAttempt
+{
+
+    /**
+     * The attempt ID, this will be a unique identifier
+     *
+     * @ORM\Id @ORM\Column(type="guid")
+     * @ORM\GeneratedValue(strategy="UUID")
+     */
+    protected $id;
+
+    /**
+     * The user ID that a login was attempted for
+     *
+     * @ORM\Column(type="integer", options={"unsigned": true})
+     */
+    protected $userId;
+
+    /**
+     * The datetime of the time the login happened. This date must be in UTC
```

Comment:
I wouldn't use a DateTime instance if we need that it's in UTC (I think that Doctrine converts it to the default PHP timezone).
If we need an UTC date/time representation I'd simply use an unsigned integer representing the UNIX timestamp.

Reply:
This is just to ensure that the input is uniform, if all entries are in the same timezone and we're selecting with the proper timezone all will be well

Reply:
As I said, when doctrine reads a DateTime field from the database, it assumes that it's in the default PHP time zone, which may not be UTC.
So, using UTC in the queries/comparisons may lead to the opposite result: you are not sure that everything is uniform.
IMHO we have 2 options: working with Unix timestamps (integers), which are absolute (they don't depend on any time zone), or we use the default PHP time zone (what I called `system` in the date helper)

Reply:
Yeah looks like it's much more complicated than I thought: https://www.doctrine-project.org/projects/doctrine-orm/en/2.6/cookbook/working-with-datetime.html. I'll just convert it to store an int

That UtcDateTimeType looks really neat.